python/nav/Snmp/pynetsnmp.py

```diff
@@ -117,19 +176,29 @@ def _build_cmdline(self):
         else:
             host = 'udp6:[%s]' % self.host if address.version() == 6 else self.host
 
-        return (
-            '-v' + self.version,
-            '-c',
-            self.community,
-            '-r',
-            str(self.retries),
-            '-t',
-            str(self.timeout),
-            '%s:%s' % (host, self.port),
+        params = [f"-v{self.version}"]
+
+        if self.version in ("1", "2c"):
+            params.extend(["-c", self.community])
+        elif self.version == "3":
+            params.extend(["-l", self.sec_level.value, "-u", self.sec_name])
+            if self.auth_protocol:
+                params.extend(["-a", self.auth_protocol.value])
+            if self.auth_password:
+                params.extend(["-A", self.auth_password])
+            if self.priv_protocol:
+                params.extend(["-x", self.priv_protocol.value])
+            if self.priv_password:
+                params.extend(["-X", self.priv_password])
+
+        params.extend(
+            ["-r", str(self.retries), "-t", str(self.timeout), f"{host}:{self.port}"]
```

Comment:
Any specific reason these need to be at the end? I'd be tempted to merge this line with line 179.

Reply:
The `-r` and `-t` arguments probably don't.  But the agent spec is usually placed last, due to what the documentation says,  e.g. `snmpcmd(1)`:
```
snmpcmd [OPTIONS] AGENT [PARAMETERS]
```

or `snmpwalk(1)`:
```
snmpwalk [APPLICATION OPTIONS] [COMMON OPTIONS] AGENT [OID]
```

(and of course, 20 years of of conventional usage of these command line programs).

The version of the commands on my current computer seem to accept the arguments in arbitrary order, however, so I can't be sure whether this was always true, or whether it was introduced at some specific version.

Reply:
You could add a comment "# stays here: do not meddle with 20 years of ingrained habits" or something :)